src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewDataDictionaryFactory.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class ViewDataDictionaryFactory
+    {
+        public static ViewDataDictionary CreateViewDataDictionary(IModelMetadataProvider metadataProvider)
```

Comment:
Hmm, good idea. If we do that and it's possible, remove the `internal` constructor entirely. Bit of a wart.